### PR TITLE
1.1 Release

### PR DESCRIPTION
**Pull request by chris-cheshire**
## Enhancements & fixes

* Updated pipeline template to nf-core/tools `2.2`
* [[#71](https://github.com/nf-core/cutandrun/issues/71)] - Bumped Nextflow version `21.04.0` -> `21.10.3`
* Added pipeline diagram to [[README](https://github.com/nf-core/cutandrun/blob/master/README.md)]
* Upgraded all modules (local and nf-core) to support the new versioning system
* The module `getchromsizes` was submitted to nf-core and moved from `local` to `nf-core`
* Added support for GFF files in IGV session generation
* [[#57](https://github.com/nf-core/cutandrun/issues/57), [#66](https://github.com/nf-core/cutandrun/issues/66)] - Upgraded version reporting in multiqc to support both software version by module and unique software versions. This improves detection of multi-version software usage in the pipeline
* [[#54](https://github.com/nf-core/cutandrun/issues/54)] - Fixed pipeline error where dots in sample ids inside the sample sheet were not correctly handled
* [[#75](https://github.com/nf-core/cutandrun/issues/75)] - Fixed error caused by emtpy peak files being passed to the `CALCULATE_FRIP` and `CALCULATE_PEAK_REPROD` python reporting modules
* [[#83]](https://github.com/nf-core/cutandrun/issues/83) - Fixed error in violin chart generation with cast to int64

### Software dependencies

Note, since the pipeline is now using Nextflow DSL2, each process will be run with its own [Biocontainer](https://biocontainers.pro/#/registry). This means that on occasion it is entirely possible for the pipeline to be using different versions of the same tool. However, the overall software dependency changes compared to the last release have been listed below for reference.

| Dependency | Old version | New version |
|------------|-------------|-------------|
| `samtools` | 1.13        | 1.14        |

> **NB:** Dependency has been __updated__ if both old and new version information is present.
> **NB:** Dependency has been __added__ if just the new version information is present.
> **NB:** Dependency has been __removed__ if version information isn't present.
## PR checklist

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
    - [x] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/cutandrun/tree/master/.github/CONTRIBUTING.md)
    - [x] If necessary, also make a PR on the nf-core/cutandrun _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint`).
- [x] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [x] Usage Documentation in `docs/usage.md` is updated.
- [x] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [x] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Review by mashehu — APPROVED** [review]
Looks good to me. A bit of cleanup needed for some remnants of the old file structure for the modules, but a `nf-core modules update` should do the trick 🙂

File: modules/nf-core/modules/bedtools/bamtobed/functions.nf [hunk 1]
```diff
@@ -9,6 +9,13 @@ def getSoftwareName(task_process) {
     return task_process.tokenize(':')[-1].tokenize('_')[0].toLowerCase()
```
[the remark this answers is not in the document]
Comment (chris-cheshire) [reply modules/nf-core/modules/bedtools/bamtobed/functions.nf:9]: We haven't implemented the new module structure yet - this is a huge job and I wanted to get these bug fixes out first.

File: modules/local/linux/awk.nf [hunk 2]
```diff
@@ -1,4 +1,4 @@
-include { initOptions; saveFiles; getSoftwareName } from '../common/functions'
+include { initOptions; saveFiles; getSoftwareName; getProcessName } from '../common/functions'
```
[the remark this answers is not in the document]
Comment (chris-cheshire) [reply modules/local/linux/awk.nf:1]: This is whats in other modules I have seen in other pipelines but I can update

File: assets/multiqc_config.yaml [hunk 3]
```diff
@@ -46,6 +46,12 @@ module_order:
     - picard
     - custom_content
 
+custom_content:
+  order:
+    - fragment_lengths
+    - software_versions
+    - software_versions_unique
+
```
[the remark this answers is not in the document]
Comment (chris-cheshire) [reply assets/multiqc_config.yaml:54]: This is great thanks

File: modules/local/linux/awk.nf (hunk 2 above)
[the remark this answers is not in the document]
Comment (chris-cheshire) [reply modules/local/linux/awk.nf:1]: Updated

File: workflows/cutandrun.nf [hunk 4]
```diff
@@ -860,6 +880,7 @@ workflow CUTANDRUN {
                 ch_seacr_bed.collect{it[1]}.ifEmpty([]),
                 UCSC_BEDGRAPHTOBIGWIG.out.bigwig.collect{it[1]}.ifEmpty([])
             )
+            // ch_software_versions = ch_software_versions.mix(IGV_SESSION.out.versions)
```
[the remark this answers is not in the document]
Comment (chris-cheshire) [reply workflows/cutandrun.nf]: This currently causes an error and is the subject of an investigation - I have commented the code to show this

**Comment by mashehu** [comment]
Oh another thing I saw when running the test: a rogue `cut` directory with `h3k27me3_R1.repro.cut.bed` inside is also generated in the results directory, which shouldn't be there I guess...

File: modules/local/common/functions.nf [hunk 5]
```diff
@@ -1,59 +1,79 @@
-/*
```
Comment (FriederikeHanssen) [inline modules/local/common/functions.nf:1]: I don't think the functions.nf is needed anymore for local modules.... see also rnaseq pipeline https://github.com/nf-core/rnaseq/blob/master/modules/local/bedtools_genomecov.nf

**Review by FriederikeHanssen — COMMENTED** [review]
Looks good, just that the new syntax without the `functions.nf` and without the `addParams` functionality is not part of this release. I am not sure what the consensus here is about switching now or leaving it here as is, since it is quite a big change.

**Review by FriederikeHanssen — APPROVED** [review]
🚀 LGTM

**Pull request merged**